### PR TITLE
Feature/logisland 511 opencv

### PR DESCRIPTION
**Pull request by oalam**

### CODE REVIEW

File: logisland-components/logisland-processors/logisland-processor-computer-vision/src/main/java/com/hurence/logisland/cv/processor/RunScript.java [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+            logger.error("Processor not initialized, returning input records and doing nothing!");
+            return records;
+        }
+
+        String imageInputField = context.getPropertyValue(INPUT_FIELD).asString();
+        String imageOutputField = context.getPropertyValue(OUTPUT_FIELD).asString();
+        String imageFormat = context.getPropertyValue(IMAGE_FORMAT).asString();
+
+        // do we replace existing records or add new ones ?
+        Collection<Record> outputRecords = new ArrayList<>(records);
```
Comment (MiniPlayer) [inline logisland-components/logisland-processors/logisland-processor-computer-vision/src/main/java/com/hurence/logisland/cv/processor/RunScript.java]: why cloning records first to eventually discard this afterwards ? This is not efficient, I would do the opposite.

File: logisland-components/logisland-processors/logisland-processor-computer-vision/src/main/java/com/hurence/logisland/cv/utils/ClojureWrapper.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+    public static void main(String[] args) throws IOException {
+        NativeLoader.loadLibrary(Core.NATIVE_LIBRARY_NAME);
+       // callLoop(true);
+        blurtheCat();
+        greyCat();
+        sepiaCat();
+        redmaskCat();
+        // Call it!
+      /*  Object result = foo.invoke("Hi", "there");
```
Comment (MiniPlayer) [inline logisland-components/logisland-processors/logisland-processor-computer-vision/src/main/java/com/hurence/logisland/cv/utils/ClojureWrapper.java]: this comment should be deleted.

File: logisland-components/logisland-processors/logisland-processor-computer-vision/src/main/java/com/hurence/logisland/cv/utils/ClojureWrapper.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.*;
+
+import static com.hurence.logisland.cv.utils.CVUtils.toBI;
+import static com.hurence.logisland.cv.utils.CVUtils.toMat;
+
+
+public class ClojureWrapper {
```
Comment (MiniPlayer) [inline logisland-components/logisland-processors/logisland-processor-computer-vision/src/main/java/com/hurence/logisland/cv/utils/ClojureWrapper.java]: What is the use of this class ? If this is just a local job for your test with Clojure this should not be commited or should be moved in test so this is not packaged in logisland.
The resources associated (images) should be moved into test resources as well.

File: logisland-components/logisland-processors/logisland-processor-computer-vision/src/main/resources/opencv.clj [hunk 4]
```diff
@@ -0,0 +1,63 @@
+; foo.clj
+(ns com.hurence.logisland
```
Comment (MiniPlayer) [inline logisland-components/logisland-processors/logisland-processor-computer-vision/src/main/resources/opencv.clj]: Does it belong to main or test ?

File: logisland-core/logisland-api/src/main/java/com/hurence/logisland/record/RecordDictionary.java [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
  * Copyright (C) 2016 Hurence (support@hurence.com)
- *
+ * <p>
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
- *
- *         http://www.apache.org/licenses/LICENSE-2.0
- *
+ * <p>
```
Comment (MiniPlayer) [inline logisland-core/logisland-api/src/main/java/com/hurence/logisland/record/RecordDictionary.java]: Seems to not be correct.

File: logisland-components/logisland-processors/logisland-processor-computer-vision/src/main/java/com/hurence/logisland/cv/processor/RunScript.java (hunk 1 above)
Comment (oalam) [reply logisland-components/logisland-processors/logisland-processor-computer-vision/src/main/java/com/hurence/logisland/cv/processor/RunScript.java]: totally right man !

File: logisland-components/logisland-processors/logisland-processor-computer-vision/src/main/java/com/hurence/logisland/cv/processor/RunScript.java [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+            throw new InitializationException(e);
+        }
+
+
+    }
+
+
+    @Override
+    public Collection<Record> process(ProcessContext context, Collection<Record> records) {
+        if (!isInitialized) {
```
[the remark this answers is not in the document]
Comment (oalam) [reply logisland-components/logisland-processors/logisland-processor-computer-vision/src/main/java/com/hurence/logisland/cv/processor/RunScript.java]: I see want you mean. let's remove this statement and see what it does in the long run ...

**Pull request merged**